### PR TITLE
Chip Rewrite

### PR DESCRIPTION
**Pull request by A-da-m**
- Switched to Eris + Quartz
- Class based commands + events
- Cleaned up current code

**Beware, breaking changes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # Chip
-Chip is the discord bot that was made for the discord server [Code Together](https://discord.gg/4uHuFTn)
+Chip is the discord bot that was made for the discord server [code::together](https://discord.gg/4uHuFTn)
```
Comment (abcdan) [inline README.md]: Can we go back to Code Together, as code::together is just the Discord name ;P

File: README.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 ### Setup:
 You require:
 - [Discord.js](https://discord.js.org/)
 - [Node-fetch](https://www.npmjs.com/package/node-fetch)
 
 Steps to setup the bot
-1. Simply rename `example-config.js` to `config.js` and changed the details to your likings.
+1. Simply rename `example.config.js` to `config.js` and changed the details to your likings.
```
Comment (Kinolite) [inline README.md]: Replace "changed" to "change" and "likings" to "liking".

File: commands/bots/accept.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+						.catch(error => {
+							return msg.embed(`**Unable to kick bot**\nError: ${error.message}`);
+						});
+				const botInviteURL = `https://discordapp.com/oauth2/authorize?client_id=${msg.channel.topic}&scope=bot&permissions=0&guild_id=${servers.main.id}`;
+				const embed = this.client.embed()
+						.title('**Invite**')
+						.description(`[URL](${botInviteURL})`)
+						.footer('Channel Deletion: 10 seconds');
+				await msg.channel.createMessage({ embed });
+				setTimeout(() => {
```
Comment (Kinolite) [inline commands/bots/accept.js]: Maybe set the timeout to be a minute? 10 seconds is too short.

File: commands/other/tag.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                content: 'Basic tags.'
+            }
+        });
+    }
+
+    async run (msg, args) {
+        await msg.delete()
+        switch (args[0]) {
+            case 'dontasktoask':
+                await msg.embed('Don\'t ask to ask. By asking to ask a question, you have already asked a question. Provide as much info as you can when asking for help.\nhttps://dontasktoask.com/');
```
Comment (Kinolite) [inline commands/other/tag.js]: Add a space after the linebreak (\n) along with a dash (-). It looks cleaner like such:
```
Don't ask to ask. ...
   - https://dontasktoask.com/
```

File: commands/other/tag.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                break;
+            case 'lookingfordevs':
+                await msg.embed('If your looking for a Developer check out <#590563326782603292>\nIf you want to add to it look at [this message](https://discordapp.com/channels/489144223099125761/590563326782603292/590563515991851038)');
```
Comment (Kinolite) [inline commands/other/tag.js]: Make "Developer" lowercase, and add a comma after it.

What do you mean by the 2nd sentence?

File: commands/system/help.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const { Command } = require('quartz');
```
Comment (Kinolite) [inline commands/system/help.js]: Please do not hardcode your commands.

File: example.config.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+module.exports= {
+    dev_id: 'id', // Used so people can't test the bot when ur coding
+    dev_mode: true, // {bool} used to easily switch between bots
+    prefix: 'prefix_here',
+    msg_to_long: false, // {bool} a bit of code that will delete a msg if it has code blocks in it then put the code in a paste and resends
```
Comment (flamableassassin) [inline example.config.js]: This isn't needed because the feature has been removed

File: README.md (hunk 1 above)
Comment (A-da-m) [reply README.md]: We have wayyy to many names The website and discord are code::together. So I'm going with that.

Comment (Kinolite) [reply README.md]: We only have two names: `Code Together` and `code::together`.

In my opinion, Code Together should be used in documentation. It's speakable and more professional. If we wanted to use `code::together`, it'd look weird in many scenarios (ex: The start of a sentence).

Comment (SpyHoodle) [reply README.md]: code::together is the brand, and the name of the Discord.
The Discord isn't called Code Together, so why call it that?

Comment (Kinolite) [reply README.md]: I prefer using `Code Together` for clarity and neatness. `code::together` is a reference to C++, but we support all languages (not like it matters).

Comment (A-da-m) [reply README.md]: I think code::together looks better. Code Together looks weird.

Comment (SpyHoodle) [reply README.md]: ^

Comment (Kinolite) [reply README.md]: @abcdan can we resolve this yet? No point in going back and forth over the name as that'll probably not be changing.

**Pull request merged**